### DISCUSSION TITLE
Bug: UI Freeze and Timeout During Long Context Interactions

### DISCUSSION THREAD

**Discussion by zhiqing0205** [open-webui/open-webui#4752] [General]
## Installation Method

Docker

## Environment

- **Open WebUI Version:** v0.3.13

- **Operating System:** Windows
- **Browser (if applicable):** Google Chrome

**Confirmation:**

- [x] I have read and followed all the instructions provided in the README.md.
- [x] I am on the latest version of both Open WebUI and Ollama.
- [x] I have included the browser console logs.
- [x] I have included the Docker container logs.
- [x] I have provided the exact steps to reproduce the bug in the "Steps to Reproduce" section below.

## Expected Behavior:

The UI should remain responsive and process requests without timeouts, even during long context interactions.

## Actual Behavior:

During long context interactions, the UI becomes unresponsive, requests time out, and the browser eventually reports a page crash.

## Description

**Bug Summary:**
We've noticed an issue where the UI becomes unresponsive and requests time out when dealing with long context interactions. This leads to a degraded user experience and eventually results in the browser reporting a page crash.

## Reproduction Details

**Steps to Reproduce:**
1. Open the OpenWebUI interface.
2. Initiate a conversation or interaction that involves a long context (please specify approximate length if possible).
3. Observe the UI becoming unresponsive.
4. Wait for the request to time out.
5. Note the continued UI freezing.
6. Eventually, the browser reports a page crash.

## Logs and Screenshots

**Browser Console Logs:**
[Please include relevant browser console logs]

**Docker Container Logs:**
[Please include relevant Docker container logs, if applicable]

**Screenshots/Screen Recordings (if applicable):**
[If possible, please attach any relevant screenshots or screen recordings that illustrate the issue]

## Additional Information

We understand that handling long contexts can be challenging, but we believe improving this aspect would greatly enhance the user experience. Any insights into potential causes or suggestions for workarounds would be greatly appreciated.

## Note

We apologize if we've overlooked any existing documentation or solutions for this issue. If there are any additional steps we should take or information we should provide to help diagnose this problem, please let us know. Thank you for your time and assistance in improving OpenWebUI!

**Comment by zeyugao** [comment]
I have experienced performance issues in Safari, when generating inside a slightly long context, the UI will be laggy.

An example chat JSON: 
[chat-export-1724131076347.json](https://github.com/user-attachments/files/16669431/chat-export-1724131076347.json)

When doing generating, CPU usage will be 100%

<img width="277" alt="image" src="https://github.com/user-attachments/assets/804140cb-b377-4c71-bc84-eee385053bf0">

This issue does not exist in Chrome, maybe the context is not long enough as mentioned before.

Open WebUI version: v0.3.13
Safari: Version 17.6 (19618.3.11.11.5)
OS: macOS 14.6.1 (23G93)

**Comment by sergio-u** [comment]
I'm also facing this issue in MS edge:

Version 128.0.2739.33 (Official build) stable app, beta channel (64-bit)

**Comment by Joakim-T** [comment]
Same issue here. This is a major issue and happens in all browser i have tested. Chrome, Edge and Firefox.
Tab completely freezes until, in this case, chatgpt 4o has sent all output.

Have tried switching on the setting  "Fluidly stream large external response chunks" but then the output is painfully slow.

**Comment by nhadsall** [comment]
Also having this problem with multiple browsers and a very fast server.

**Comment by GrayXu** [comment]
[#3620](https://github.com/open-webui/open-webui/issues/3620)
